packages/block-library/src/navigation/edit/navigation-menu-selector.js

```diff
@@ -43,24 +41,6 @@ function NavigationMenuSelector(
 		canSwitchNavigationMenu,
 	} = useNavigationMenu();
 
-	const handleSelect = useCallback(
-		( _onClose ) => ( selectedId ) => {
-			_onClose();
-			onSelect(
-				navigationMenus?.find( ( post ) => post.id === selectedId )
-			);
-		},
-		[ navigationMenus ]
-	);
-
-	const handleSelectClassic = useCallback(
-		( _onClose, menu ) => () => {
-			_onClose();
-			onSelect( menu );
-		},
-		[]
-	);
-
```

Comment:
I don't think these need to be memoized, so I removed that. It turns out that the `navigationMenus?.find( ( post ) => post.id === selectedId )` isn't needed as the `onSelect` callback only uses the `id` of the menu.

The new `onSelectNavigationMenu` prop can be passed directly to `MenuItemsChoice`